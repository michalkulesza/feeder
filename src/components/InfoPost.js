import React from "react";

const InfoPost = () => {
  return (
    <div className="post info-post">
      <p>Please log in to see the feed.</p>
    </div>
  );
};

export default InfoPost;
