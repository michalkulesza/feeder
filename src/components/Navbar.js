import React from "react";
import "./Navbar.scss";
import { withRouter, Link } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { logoutUser } from "../redux/actions/userAction";

const button = (props, authUid, dispatch, userName) => {
  switch (props.location.pathname) {
    case "/signup":
      return (
        <Link to="/login">
          <div className="button">Login</div>
        </Link>
      );
    case "/login":
      return (
        <Link to="/signup">
          <div className="button">Signup</div>
        </Link>
      );
    default: {
      return authUid && userName ? (
        <>
          <Link to={`/user/${userName}`}>
            <span>{userName}</span>
          </Link>
          <div
            className="button red"
            onClick={() => {
              dispatch(logoutUser(props.history));
            }}
          >
            Logout
          </div>
        </>
      ) : (
        <Link to="/login">
          <div className="button">Login</div>
        </Link>
      );
    }
  }
};

const Navbar = props => {
  const authUid = useSelector(state => state.firebase.auth.uid);
  const userName = useSelector(state => state.firebase.profile.username);
  const dispatch = useDispatch();

  return (
    <div className="navbar-container">
      <div className="wrapper">
        <Link to="/">
          <div className="logo">Feeder</div>
        </Link>
        {button(props, authUid, dispatch, userName)}
      </div>
    </div>
  );
};

export default withRouter(Navbar);
