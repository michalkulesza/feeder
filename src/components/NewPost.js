import React, { useState, useEffect } from "react";
import "./NewPost.scss";
// import TextareaAutosize from "react-textarea-autosize";
import { useDispatch } from "react-redux";
import { addPost } from "../redux/actions/postActions";
import { Editor, EditorState } from "draft-js";
import { stateToHTML } from "draft-js-export-html";

import PostMisc from "./PostMisc";
import avatar from "../res/avatar-default.png";

const NewPost = ({ uid, username, users }) => {
  const dispatch = useDispatch();
  const postsInitValue = {
    body: "",
    createdAt: undefined,
    author: username,
    uid,
    likes: 0,
    likesUsers: [],
    likesUid: []
  };

  const [isFocus, setIsFocus] = useState(false);
  const [active, setActive] = useState(false);
  const [post, setPost] = useState(postsInitValue);
  const [editorState, setEditorState] = useState(EditorState.createEmpty());
  const [editorContentInHTML] = useState(null);
  let editor = React.createRef();

  useEffect(() => {
    isFocus ? setActive(true) : setActive(false);
  }, [isFocus]);

  const onEditorChange = editorState => {
    setEditorState(editorState);
    setPost({
      ...postsInitValue,
      body: stateToHTML(editorState.getCurrentContent()),
      createdAt: new Date().toISOString()
    });
    console.log(post.body);
  };

  const handleNewPost = () => {
    post.body.trimLeft();

    if (post.body !== "" && post.createdAt && post.author && post.uid) {
      let currentUserPosts = users[uid].posts;
      dispatch(addPost(post, currentUserPosts));
      setPost(postsInitValue);
      setEditorState(EditorState.createEmpty());
    }
  };

  return (
    <div className="post">
      <div className="post-header">
        <div className="post-avatar">
          <img src={avatar} alt="" />
        </div>
        <div className="post-body new-post-body">
          <Editor
            editorState={editorState}
            onChange={onEditorChange}
            placeholder="Say something..."
            onFocus={() => setIsFocus(true)}
            onBlur={() => setIsFocus(false)}
            ref={editor}
          />
        </div>
      </div>
      <div dangerouslySetInnerHTML={editorContentInHTML}></div>
      <PostMisc active={active} handleNewPost={handleNewPost} />
    </div>
  );
};

export default NewPost;
