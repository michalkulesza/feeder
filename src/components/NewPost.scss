@import "../main.scss";

.new-post-body {
  width: 100%;
}

#textarea {
  padding: 1rem;
  font-size: 0.9rem;
  width: 100%;
  resize: none;
  border: 1px solid $lighterGray;
  overflow: hidden;
  border-radius: 5px;
}
