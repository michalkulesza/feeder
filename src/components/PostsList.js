import React from "react";
import Post from "./Post";

function PostsList({ posts }) {
  return (
    <div>
      {posts.length > 0
        ? posts.map(post => (
            <Post
              content={post.body}
              author={post.userName}
              createdAt={post.createdAt}
              key={post.id}
            />
          ))
        : ""}
    </div>
  );
}

export default PostsList;
