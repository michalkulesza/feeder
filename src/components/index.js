export { default as InfoPost } from "./InfoPost/InfoPost";
export { default as Login } from "../pages/Login/Login";
export { default as Navbar } from "./Navbar/Navbar";
export { default as Post } from "./Post/Post";
export { default as PostMisc } from "./PostMisc/PostMisc";
export { default as PostsList } from "./PostsList/PostsList";
export { default as PostSocial } from "./PostSocial/PostSocial";
export { default as SignUp } from "../pages/SignUp/SignUp";
export { default as UserDetails } from "./UserDetails/UserDetails";
