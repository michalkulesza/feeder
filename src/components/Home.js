import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { useFirestoreConnect } from "react-redux-firebase";
import { deletePost } from "../redux/actions/postActions";

import PostsList from "./PostsList";
import InfoPost from "./InfoPost";
import NewPost from "./NewPost";

const Home = () => {
  useFirestoreConnect([{ collection: "posts", orderBy: [["createdAt", "desc"]] }]);

  const dispatch = useDispatch();
  const uid = useSelector(state => state.firebase.auth.uid);
  const posts = useSelector(state => state.firestore.ordered.posts);
  const username = useSelector(state => state.firebase.profile.username);
  const handleDeletePost = key => {
    dispatch(deletePost(key));
  };

  return (
    <div className="container">
      <div className="home wrapper">
        {uid && posts ? (
          <>
            <NewPost uid={uid} username={username} />
            <PostsList
              posts={posts}
              uid={uid}
              handleDelete={handleDeletePost}
              username={username}
            />
          </>
        ) : uid && !posts ? (
          <>
            <NewPost uid={uid} username={username} />
            <InfoPost>There are no posts yet.</InfoPost>
          </>
        ) : (
          <InfoPost>Please log in to see the feed.</InfoPost>
        )}
      </div>
    </div>
  );
};

export default Home;
