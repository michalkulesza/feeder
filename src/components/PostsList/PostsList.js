import React from "react";
import { Post } from "../../components";

const PostsList = ({ posts, uid, handleDelete, username }) => {
	return posts
		? posts.length > 0 &&
				posts.map(post => (
					<Post
						content={post.body}
						author={post.author}
						authorsUid={post.uid}
						postId={post.id}
						likes={post.likes}
						likesUid={post.likesUid}
						likesUsers={post.likesUsers}
						currentUid={uid}
						createdAt={post.createdAt}
						key={post.id}
						postKey={post.id}
						handleDelete={handleDelete}
						username={username}
					/>
				))
		: null;
};

export default PostsList;
