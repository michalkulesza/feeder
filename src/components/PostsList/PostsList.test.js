import React from "react";
import { shallow } from "enzyme";

import PostsList from "./PostsList";
import { Post } from "../../components";

describe("<PostsList/>", () => {
	const posts = [
		{ author: "", createdAt: "", content: "", id: 1223213213 },
		{ author: "", createdAt: "", content: "", id: 2323232332 },
	];

	it("should match the snapshot without props", () => {
		const wrapper = shallow(<PostsList />);

		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should render posts", () => {
		const wrapper = shallow(<PostsList posts={posts} />);

		expect(wrapper.find(Post).length).toBe(2);
	});
});
