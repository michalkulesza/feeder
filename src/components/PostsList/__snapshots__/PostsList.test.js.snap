// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsList/> should match the snapshot without props 1`] = `""`;
